src/LagrangianParticleTracking/calculate_particle_tendency_contributions.jl

```diff
@@ -0,0 +1,75 @@
+function calculate_particle_tendency_contributions!(particles, model, active_properties)
+    arch = model.architecture
+
+    workgroup = min(length(particles), 256)
+    worksize = length(particles)
+
+    events = []
+
+    for field in active_properties
+        property = getproperty(particles.properties, field.property)
+        tendency = getproperty(model.timestepper.Gⁿ, field.tracer)
+
+        Gp_kernal! = calculate_particle_tendency_kernel!(device(arch), workgroup, worksize)
+        Gp_event = Gp_kernal!(property, tendency, particles, model.tracers, model.grid)
+
+        push!(events, Gp_event)
+    end
+
+    wait(device(arch), MultiEvent(Tuple(events)))
+
+    return nothing
+end
+
+calculate_particle_tendency_contributions!(model) = calculate_particle_tendency_contributions!(model.particles, model)
+calculate_particle_tendency_contributions!(particles::Nothing, model) = nothing
+calculate_particle_tendency_contributions!(particles, model) = !isnothing(particles.parameters) ? (:active_properties in keys(particles.parameters) ? calculate_particle_tendency_contributions!(particles, model, particles.parameters.active_properties) : nothing) : nothing
+calculate_particle_tendency_contributions!(particles, model, active_fields::Nothing) = nothing
+
+@inline function getmask(x, y, z, grid::AbstractGrid{FT, TX, TY, TZ}, loc) where {FT, TX, TY, TZ}
```

Comment:
```suggestion
@inline function get_mask(x, y, z, grid::AbstractGrid{FT, TX, TY, TZ}, loc) where {FT, TX, TY, TZ}
```

A more descriptive function name might help code-readers understand this more quickly!